Framework/DataHandling/src/LoadNXcanSAS.cpp

```diff
@@ -128,6 +129,62 @@ void loadLogs(H5::Group &entry, const Mantid::API::MatrixWorkspace_sptr &workspa
   }
 }
 
+// ----- SAMPLE
+
+namespace {
+bool hasAperture(H5::Group &entry) {
```

Comment:
Would it be worth getting these methods that are checking if groups exist to return the group if it's found, so that we can avoid repeating the call to open the group again afterwards? I noticed the code further on uses `&&`, which I've read a bit about but am not very familiar with, so apologies if it should be obvious to me why this is more optimal!

Another C++ general info question, what were the reasons for defining these in a namespace rather than as private methods?